### PR TITLE
931: Crop Coordinates Error

### LINKED ISSUE #931: Crop Coordinates error if image is smaller than 200x200
### Summary

Opening the operations window with images smaller than 200x200 leads to an error. I used the "tilt1_ready2" dataset.

### Expected Behaviour

The starting ROI in crop coordinates is smaller than or equal to the size of the image.

### Current Behaviour

The starting ROI in crop coordinates is 0, 0, 200, 200.

### Context

mantidimaging 2.0.0rc_648
Ubuntu 20.04.2 LTS x86_64

### Failure Logs

```
eview: could not broadcast input array from shape (1,79,200) into shape (1,200,200)
Traceback (most recent call last):
  File "/home/work/mantidimaging/mantidimaging/gui/windows/operations/presenter.py", line 258, in do_update_previews
    self.model.apply_to_images(subset)
  File "/home/work/mantidimaging/mantidimaging/gui/windows/operations/model.py", line 120, in apply_to_images
    exec_func(images)
  File "/home/work/mantidimaging/mantidimaging/core/operations/crop_coords/crop_coords.py", line 70, in filter_func
    images.data = execute_single(sample, region_of_interest, progress, out=output)
  File "/home/work/mantidimaging/mantidimaging/core/operations/crop_coords/crop_coords.py", line 116, in execute_single
    output[:] = data[:, roi.top:roi.bottom, roi.left:roi.right]
ValueError: could not broadcast input array from shape (1,79,200) into shape (1,200,200)
Traceback (most recent call last):
  File "/home/work/mantidimaging/mantidimaging/gui/windows/operations/presenter.py", line 258, in do_update_previews
    self.model.apply_to_images(subset)
  File "/home/work/mantidimaging/mantidimaging/gui/windows/operations/model.py", line 120, in apply_to_images
    exec_func(images)
  File "/home/work/mantidimaging/mantidimaging/core/operations/crop_coords/crop_coords.py", line 70, in filter_func
    images.data = execute_single(sample, region_of_interest, progress, out=output)
  File "/home/work/mantidimaging/mantidimaging/core/operations/crop_coords/crop_coords.py", line 116, in execute_single
    output[:] = data[:, roi.top:roi.bottom, roi.left:roi.right]
ValueError: could not broadcast input array from shape (1,79,200) into shape (1,200,200)
```

### Screenshot(s)

image

### PR DESCRIPTION
**Pull request by DolicaAkelloEgwel**
### Issue

Closes #931 

### Description

Sets the initial crop coordinates value to something smaller than the current image.

### Testing 

Added tests for the `init_cop_coords` function.

### Acceptance Criteria 

Load the  "tilt1_ready2" dataset and open the operations window. Check that no error message appears.

### Documentation

Updated the release notes.

### CODE REVIEW

File: mantidimaging/gui/windows/operations/test/test_presenter.py [hunk 1]
```diff
@@ -361,3 +365,23 @@ def test_running_operation_records_previous_button_states(self, _):
...
+        mock_roi_field.setText.assert_not_called()
+
+    def test_init_crop_coords_does_nothing_when_image_is_greater_than_200_by_200(self):
+        mock_roi_field = mock.Mock()
+        self.presenter.stack = mock.Mock()
+        self.presenter.stack.presenter.images.data = np.ones((2, 201, 201))
+        self.presenter.init_crop_coords(mock_roi_field)
+        mock_roi_field.setText.assert_not_called()
+
+    @parameterized.expand([(190, 201, "0, 0, 200, 95"), (201, 80, "0, 0, 40, 200"), (200, 200, "0, 0, 100, 100")])
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/operations/test/test_presenter.py:381]: Could this be done with pytest to save needing a new dependency? https://docs.pytest.org/en/stable/parametrize.html

File: mantidimaging/gui/windows/operations/presenter.py [hunk 2]
```diff
@@ -324,3 +329,23 @@ def _set_apply_buttons_enabled(self, apply_single_enabled: bool, apply_all_enabl
...
+            return
+
+        larger = np.greater(self.stack.presenter.images.data[0].shape, (200, 200))
+        if all(larger):
+            return
+        x = y = 200
+        if not larger[0]:
+            x = self.stack.presenter.images.data[0].shape[0] // 2
+        if not larger[1]:
+            y = self.stack.presenter.images.data[0].shape[1] // 2
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/operations/presenter.py:349]: Would this be clearer as

```
x = min(self.stack.presenter.images.data[0].shape[0], 200)
y = min(self.stack.presenter.images.data[0].shape[1], 200)
```

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]

File: mantidimaging/gui/windows/operations/test/test_presenter.py (hunk 1 above)
Comment (DolicaAkelloEgwel) [reply mantidimaging/gui/windows/operations/test/test_presenter.py:381]: This doesn't seem to work when the class inherits from `unittest.TestCase`.

Comment (samtygier-stfc) [reply mantidimaging/gui/windows/operations/test/test_presenter.py:381]: ah, ok.

**Review by samtygier-stfc — APPROVED** [review]

**Pull request merged**